game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
+                                    build_text(&msgs, &norm_distance),
+                                ));
+
+                                let mut display_txt = Text::from(Line(&p.phase_type.describe(map)));
+                                display_txt.add(Line(format!("  Intersection: {}", turn.parent)));
+                                display_txt.add(Line(format!(
+                                    "  Delay: {}",
+                                    Duration::seconds(*delay as f64)
+                                )));
+                                segments.push((
+                                    // To make sure that the hotspot isn't too small
+                                    if 0.05 < (turn_details.geom.length() / sum_phase_dist) {
+                                        turn_details.geom.length()
+                                    } else {
+                                        sum_phase_dist += sum_phase_dist * 0.05;
+                                        sum_phase_dist -= turn_details.geom.length();
+                                        sum_phase_dist * 0.05
+                                    },
+                                    Color::RED,
+                                    display_txt,
+                                ));
+                                norm_distance = Distance::meters(0.0);
+                                found = true;
+                                break;
+                            }
+                        }
+                        if !found {
+                            norm_distance += turn_details.geom.length();
+                        }
+                    }
+                }
+            }
+            segments.push((norm_distance, norm_color, build_text(&msgs, &norm_distance)));
+        } else {
+            // TODO Think of something to do instead
+            unimplemented!("We dont have a path!!");
+        }
+        icon_pos.push(sum_phase_dist / 2.0);
+        total_dist += sum_phase_dist;
+    }
+    let mut timeline = Vec::new();
+    let mut icons = Vec::new();
+    assert_eq!(icons_geom.len(), icon_pos.len());
+    let mut offset = 0.0;
+    for index in 0..icon_pos.len() {
+        let pos = box_width * (*icon_pos.get(index).unwrap() / total_dist);
+        let img = icons_geom.get(index).unwrap().to_owned();
+        let img_size = img.get_dims();
+        icons.push(Widget::draw_batch(
+            ctx,
+            img.centered_on(
+                // TODO Hardcoded layouting...
+                Pt2D::new(pos + offset, -20.0),
+            ),
+        ));
+        offset += (pos * 2.0) - img_size.width;
+    }
+    for seg in segments {
+        let seg_percent = seg.0 / total_dist;
+        let seg_with = seg_percent * box_width;
+        let rect = Polygon::rectangle(seg_with, 15.0);
+        let batch = GeomBatch::from(vec![(seg.1, rect.clone())]);
+        timeline.push(
+            DrawWithTooltips::new(
```

Comment:
Woops, I guess when I made `DrawWithTooltips`, I didn't realize the entire timeline could just be one of these. I'll make that cleanup after your PR. There's hopefully some simplifications that can happen here.